okhttp/src/jvmTest/java/okhttp3/internal/idn/PlainTextIdnaMappingTable.kt

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal.idn
+
+import okio.Buffer
+import okio.BufferedSink
+import okio.BufferedSource
+import okio.ByteString
+import okio.ByteString.Companion.encodeUtf8
+import okio.IOException
+import okio.Options
+
+/**
+ * A decoded [mapping table] that can perform the [mapping step] of IDNA processing.
+ *
+ * This implementation is optimized for readability over efficiency.
+ *
+ * [mapping table]: https://www.unicode.org/reports/tr46/#IDNA_Mapping_Table
+ * [mapping step]: https://www.unicode.org/reports/tr46/#ProcessingStepMap
+ */
+class PlainTextIdnaMappingTable internal constructor(
+  private val mappings: List<Mapping>,
+) : IdnaMappingTable {
+  override fun apply(codePoint: Int, sink: BufferedSink): Boolean {
+    val index = mappings.binarySearch {
+      when {
+        it.sourceCodePoint1 < codePoint -> -1
+        it.sourceCodePoint0 > codePoint -> 1
+        else -> 0
+      }
+    }
+
+    val mapping = mappings[index]
```

Comment:
Should you handle missing values before an IOOBE?

Reply:
YES! Done.